apps/prairielearn/src/pages/publicQuestionFileBrowser/publicQuestionFileBrowser.ts

```diff
@@ -0,0 +1,77 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+import fs from 'fs-extra';
+
+import * as error from '@prairielearn/error';
+
+import { FileBrowser, browseDirectory, browseFile } from '../../components/FileBrowser.html.js';
+import { UserSchema } from '../../lib/db-types.js';
+import { getPaths } from '../../lib/instructorFiles.js';
+import { encodePath } from '../../lib/uri-util.js';
+import { selectCourseById } from '../../models/course.js';
+import { selectQuestionById } from '../../models/question.js';
+
+const router = Router({ mergeParams: true });
+
+async function setLocals(req, res) {
+  res.locals.user = UserSchema.parse(res.locals.authn_user);
+  res.locals.authz_data = { user: res.locals.user };
+  res.locals.course = await selectCourseById(req.params.course_id);
+  res.locals.question = await selectQuestionById(req.params.question_id);
+
+  if (
+    !res.locals.question.share_source_publicly ||
+    res.locals.course.id !== res.locals.question.course_id
+  ) {
+    throw new error.HttpStatusError(404, 'Not Found');
+  }
+  return;
```

Comment:
This is unnecessary; a return at the end of a function is implicit.

As with the other file, there's really no need for this to be a separate function, it can be inlined in the `GET` handler. That said, maybe there's an argument that some or all of this could be lifted into reusable middlewares that we could share across multiple pages. I haven't thought about this too deeply.